File: app/src/main/java/com/prashant/masterbuddy/room/AppDatabase.java
package com.prashant.masterbuddy.room;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

@Database(entities = {Video.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase instance;

    public abstract VideoDao videoDao();

    public static AppDatabase getAppDatabase(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context, AppDatabase.class, "database")
                    .allowMainThreadQueries().build();
        }
        return instance;
    }

    public static void destroyInstance() {
        instance = null;
    }
}

File: app/src/main/java/com/prashant/masterbuddy/room/AppDatabase.kt
package com.prashant.masterbuddy.room

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

@Database(entities = [Video::class, SavedMedia::class], version = 1)
abstract class AppDatabase : RoomDatabase() {

    abstract fun videoDao(): VideoDao

    abstract fun savedMediaDao(): SavedMediaDao

    companion object {

        private var instance: AppDatabase? = null

        fun getAppDatabase(context: Context): AppDatabase {
            if (instance == null) {
                instance = Room.databaseBuilder<AppDatabase>(context, AppDatabase::class.java, "database")
                        .allowMainThreadQueries().build()
            }
            return instance!!
        }

        fun destroyInstance() {
            instance = null
        }
    }
}
